src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1882,8 +1882,13 @@ bool ReplicatedMergeTreeQueue::tryFinalizeMutations(zkutil::ZooKeeperPtr zookeep
 
     /// We need to check committing block numbers and new parts which could be committed.
     /// Actually we don't need most of predicate logic here but it all the code related to committing blocks
-    /// and updatading queue state is implemented there.
-    auto merge_pred = getMergePredicate(zookeeper);
+    /// and updatating queue state is implemented there.
+    PartitionIdsHint partition_ids_hint;
+    for (const auto & candidate : candidates)
+        for (const auto & partitions : candidate->block_numbers)
+            partition_ids_hint.insert(partitions.first);
+
+    auto merge_pred = getMergePredicate(zookeeper, /* partition_ids_hint */ {});
```

Comment:
did you want to send `partition_ids_hint` here? Because you create it and fill it up but don't use it anywhere.